lib/puppet/pops/evaluator/compare_operator.rb

```diff
@@ -222,7 +222,7 @@ def match_Hash(hash, left, scope)
 
   def match_Symbol(symbol, left, scope)
     return true if symbol == :default
-    equals(left, default, scope)
```

Comment:
This is still present in master at https://github.com/puppetlabs/puppet/blob/master/lib/puppet/pops/evaluator/compare_operator.rb#L248